cpp/src/knn/knn_mg.cu

```diff
@@ -53,36 +52,36 @@ void reduce(Matrix::Data<int64_t> *&out_I, Matrix::floatData_t *&out_D,
     cur_batch_size, idxRanks.size(), k, stream, trans.data());
 }
 
-void perform_local_knn(int64_t *res_I, float *res_D,
-                       std::vector<Matrix::floatData_t *> &idx_data,
-                       Matrix::PartDescriptor &idx_desc,
-                       std::vector<Matrix::RankSizePair *> &local_idx_parts,
-                       std::vector<size_t> &start_indices, cudaStream_t stream,
-                       cudaStream_t *internal_streams, int n_internal_streams,
-                       std::shared_ptr<deviceAllocator> allocator,
-                       size_t cur_batch_size, int k, float *cur_query_ptr,
-                       bool rowMajorIndex, bool rowMajorQuery) {
-  std::vector<float *> ptrs(idx_data.size());
-  std::vector<int> sizes(idx_data.size());
-
-  for (int cur_idx = 0; cur_idx < idx_data.size(); cur_idx++) {
-    ptrs[cur_idx] = idx_data[cur_idx]->ptr;
-    sizes[cur_idx] = local_idx_parts[cur_idx]->size;
-  }
-
-  // PartDescriptor uses size_t while FAISS uses int64_t
-  // so we need to do a quick conversion.
-  std::vector<int64_t> start_indices_long;
-  for (size_t start_index : start_indices)
-    start_indices_long.push_back((int64_t)start_index);
-
-  // ID ranges need to be offset by each local partition's
-  // starting indices.
-  MLCommon::Selection::brute_force_knn(
-    ptrs, sizes, (int)idx_desc.N, cur_query_ptr, (int)cur_batch_size, res_I,
-    res_D, k, allocator, stream, internal_streams, n_internal_streams,
-    rowMajorIndex, rowMajorQuery, &start_indices_long);
-}
+// void perform_local_knn(int64_t *res_I, float *res_D,
```

Comment:
I'm guessing this was for testing?

Reply:
Actually this symbol was defined multiple times. Should have been removed instead of commented.